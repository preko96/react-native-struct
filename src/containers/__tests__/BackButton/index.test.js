import React from 'react'
import { BackButton } from '../../BackButton/'

import configureMockStore from 'redux-mock-store'
import thunkMiddleware from 'redux-thunk'
export const mockStore = configureMockStore([thunkMiddleware])

it('renders without crashing', () => {
	expect(shallow(<BackButton />)).toMatchSnapshot()
})

it('calls onPressBackButton with the right props on click', () => {
	const fn = jest.fn()
	const expected = 'called with this value'
	const wrapper = shallow(<BackButton routeName={expected} onPressBackButton={fn}/>)
	wrapper.simulate('Press')
	expect(fn).toHaveBeenCalledTimes(1)
	expect(fn).toBeCalledWith(expected)
})

it('renders with prevRout', () => {
	expect(shallow(<BackButton prev={true} />)).toMatchSnapshot()
})