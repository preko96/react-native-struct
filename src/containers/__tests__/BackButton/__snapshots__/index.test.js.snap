// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with prevRout 1`] = `
<Styled(Button)
  onPress={[Function]}
  style={
    Object {
      "height": "100%",
    }
  }
  transparent={true}
>
  <Styled(Icon)
    name="arrow-back"
    style={
      Object {
        "color": "#007aff",
      }
    }
  />
  <Styled(Text) />
</Styled(Button)>
`;

exports[`renders without crashing 1`] = `
<Styled(Button)
  onPress={[Function]}
  style={
    Object {
      "height": "100%",
    }
  }
  transparent={true}
>
  <Styled(Icon)
    name="arrow-back"
    style={
      Object {
        "color": "#007aff",
      }
    }
  />
  <Styled(Text) />
</Styled(Button)>
`;
